modules/directives/src/main/scala/scala/build/preprocessing/directives/DirectiveHandler.scala

```diff
@@ -15,10 +27,136 @@ trait DirectiveHandler[T] {
+    */
+  protected def getSupportedTypes(key: String): Set[UsingDirectiveValueKind] =
+    Set(UsingDirectiveValueKind.STRING)
+
+  /** by default, it is `(lower = 1, upper = Int.MaxValue)`. Please, override for other cases.
+    * @param key
+    *   the using directive key for which the value number bounds are calculated.
+    */
+  protected def getValueNumberBounds(key: String): UsingDirectiveValueNumberBounds =
+    UsingDirectiveValueNumberBounds()
+
+  /** The default implementation just asks the user to put quotation marks around the using
+    * directive values with string type.
+    *
+    * Therefore, it should get overrided if other information is to be shown instead in any
+    * particular handler.
+    *
+    * This method gets internally called in the implementation of
+    * [[siftSupportedValueTypesWithNoResidue]] for producing the error value.
+    * @param key
+    *   the using directive key for which the unexpected value(s) are passed.
+    * @return
+    *   the hint to be shown to the user if the value(s) passed to the corresponding using directive
+    *   key are unexpected.
+    */
+  protected def unexpectedValueHint(key: String): String =
+    s"Did you forget to put the quotation marks around the string values passed to the using directive key $key?"
+
+  /** checks if the passed in `scopedDirective` has the expected number of values according to the
+    * implementation of [[getValueNumberBounds]] for the handler on which this is called
+    *
+    * @return
+    *   Either a [[UsingDirectiveExpectationError]] or the same passed in scopedDirective
+    */
+  final protected def receiveTheRightNumberOfValues(
+    scopedDirective: ScopedDirective
+  ): Either[UsingDirectiveExpectationError, ScopedDirective] = {
+    val values: Seq[Value[_]] = scopedDirective.directive.values.collect {
+      case n: NumericValue => n
+      case b: BooleanValue =>
+        b
+      case s: StringValue => s
+    } // TODO or perhaps `.filterNot(_.isInstanceOf[EmptyValue])` ??
+    val length       = values.length
+    val numberBounds = getValueNumberBounds(scopedDirective.directive.key)
+
+    if (length < numberBounds.lower || length > numberBounds.upper)
+      Left(new UsingDirectiveValueNumError(
+        scopedDirective.maybePath,
+        scopedDirective.directive.key,
+        numberBounds,
+        length
+      ))
+    else Right(scopedDirective.copy(directive = scopedDirective.directive.copy(values = values)))
+  }
+
+  /** It checks the values ascribed to the key inside the passed in `scopedDirective`, in order to
+    * see if they are of the types listed to be expected by the implementation of
+    * [[getSupportedTypes]] inside the specific handler.
+    *
+    * @return
+    *   Either a [[UsingDirectiveExpectationError]] or the using directive values of the
+    *   `scopedDirective` grouped in `string`, `boolean`, or `numeric` sequences.
+    */
+  final protected def siftSupportedValueTypesWithNoResidue(
```

Comment:
`siftSupportedValueTypesWithNoResidue ` describes what this methods does however it is a bit hard to understand at first glance. Maybe something long the line of `checkAndGroupValuesByType`?